### ISSUE TITLE
Factor out record type from QNAME format

### ISSUE THREAD

**Issue by enygren** [ietf-wg-dnsop/draft-ietf-dnsop-domain-verification-techniques#127]
The have the sections of the current draft factored conflates a few topics.  
In-particular, the "CNAME Records" section conflates the issue of a CNAME record type with the token being in the QNAME.
Some of the recommendations in that section are also inconsistent with recommendations elsewhere in the draft (including some things that have issues raised during review as security considerations).
Based on a re-read of the current draft, I think it could benefit from some reorganization of this.

I'd propose that in our recommendations we highlight (where "top-level" still means under section 5):

* Validation records have a qname and a target.  Tokens must be present in at least one of these.
* General QNAME construction
* TXT records containing a token
* CNAME containing the target in the QNAME  (note that we may want to have a recommendation here like <token>._foo-challenge that is consistent with other recommendations)
* Delegated DCV (as a top-level item, as while it involves a CNAME it is typically with TXT record in the end)
* Supporting multiple providers as a top-level item rather than under CNAME
* We might discourage DNAME

I can propose a PR with this refactoring.

**Comment by moonshiner** [comment]
Please Do !! I will gladly review and will poke others as well

**Comment by moonshiner** [comment]
So thinking on this - I like this a couple of Qs

do Random Token and TTL Considerations sray under Validation Record Format ? and Name becomes QNAME and there is a section on construction?  

I do think Random Token has ended up with a bunch of stuff

We should not merge my 1464 suggestions until later

**Comment by enygren** [comment]
Working on a PR for this now, at least for a first pass --- it does re-order a bunch of stuff so may conflict with various other PRs

**Comment by enygren** [comment]
PR created

**Comment by moonshiner** [comment]
My one thing - I see you use "Validation Record" but the definition is "Validation record". I also see "validation record" in other places. can you look?

**Comment by enygren** [comment]
Should we make "Validation Record" as consistent?  I can submit this into the PR but it may make it even harder to diff...

**Comment by moonshiner** [comment]
My opinions is that "if we define it, we must be consistent with its usage"

A followup PR then
